openmicroscopy/bioformats
# specification: Consolidate schema into single file

- Move BinaryFile, SA, SPW and ROI schemas into OME schema
- Upgrade and downgrade transforms should handle the namespace migration (but I'm no XSLT expert and am not 100% sure this is correct--needs testing)
- xsdfu is invoked with a single schema
- delete unused schemas and drop from catalogue
- drop/update namespace prefixes
- change tabs to spaces in the schema

xsd-fu can still process multiple schemas--needed for C++/2013-06 until we can upgrade to 2015-01.

--------

Outstanding questions:

- [ ] Is the change to OMEXMLWriter appropriate; does it need to care about multiple  model versions here?
- [x] Why is this not now generating proper BinData methods in the metadata code?  The generator bug *should* have been fixed, unless we're running into additional problems (ordering within the schema?)  That it would work immediately is probably wishful thinking on my part.  What else needs changing?
- [ ] What should the version number on this be?

--------

Testing:

- check generated source diff
- check changes to ome.xsd, ignoring whitespace changes
- test upgrade and downgrade, in particular the migration to the OME namespace on upgrade and splitting back into separate namespaces on downgrade

--------

/cc @mtbc @dgault @joshmoore
